prog/ai/inference_endpoint_replica_nexus.rb

```diff
@@ -163,33 +163,28 @@ def resolve_page
 
   # pushes latest config to inference gateway and collects billing information
   def ping_gateway
-    projects = if inference_endpoint.is_public
-      Project.where(DB[:api_key]
+    eligible_projects_ds = Project.dataset
+    eligible_projects_ds = eligible_projects_ds.where(id: inference_endpoint.project.id) unless inference_endpoint.is_public
+
+    eligible_projects = eligible_projects_ds.where(DB[:api_key]
       .where(owner_table: "project")
       .where(used_for: "inference_endpoint")
       .where(is_valid: true)
       .where(owner_id: Sequel[:project][:id]).exists)
```

Comment:
nit: The nested 'where' clauses make the query difficult to read. We could consider separating the api key dataset.

```
    api_key_ds = DB[:api_key]
      .where(owner_table: "project")
      .where(used_for: "inference_endpoint")
      .where(is_valid: true)
      .where(owner_id: Sequel[:project][:id])
      .exists

    eligible_projects_ds = Project.where(api_key_ds)
    eligible_projects_ds = eligible_projects_ds.where(id: inference_endpoint.project.id) unless inference_endpoint.is_public


```

Reply:
updated according to your suggestion